ext/dom/node.c

```diff
@@ -1220,6 +1220,9 @@ PHP_METHOD(DOMNode, appendChild)
 	}
 
 	if (new_child == NULL) {
+		if (!child->ns && nodep->nsDef) {
```

Comment:
I'm not sure whether this shouldn't be:
```suggestion
		if (!child->ns && nodep->ns) {
```
That would solve:
````php
$dom = new \DOMDocument();
$dom
  ->appendChild($dom->createElementNS('some:namespace', 'foo'))
  ->appendChild($dom->createElement('bar'))
  ->appendChild($dom->createElement('baz'));
echo ($xml = $dom->saveXML());

$xpath = new \DOMXPath($dom);
$xpath->registerNamespace('n', 'some:namespace');
echo count($xpath->query('/n:foo/bar')) . " should be 0\n";
echo count($xpath->query('/n:foo/n:bar')) . " should be 1\n";
echo count($xpath->query('/n:foo/n:bar/baz')) . " should be 0\n";
echo count($xpath->query('/n:foo/n:bar/n:baz')) . " should be 1\n\n";
````

Reply:
nodep->ns is not a nullptr but point to a invalid address If nodep is the root node.